src/App.jsx

```diff
@@ -14,33 +15,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [activeSort, setActiveSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlphabetically = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.localeCompare(b));
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    setGoods(sortedGoods);
+    setActiveSort('alphabetical');
+    setIsReversed(false);
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortByLength = () => {
+    const sortedGoods = [...goods].sort((a, b) => a.length - b.length);
+
+    setGoods(sortedGoods);
+    setActiveSort('length');
+    setIsReversed(false);
+  };
+
+  const reverseSort = () => {
+    const reversedGoods = [...goods].reverse();
+
+    setGoods(reversedGoods);
+    setIsReversed(!isReversed);
+    setActiveSort('');
+  };
+
+  const resetSort = () => {
+    setGoods(goodsFromServer);
+    setActiveSort('');
+    setIsReversed(false);
+  };
+
+  const isResetVisible = goods !== goodsFromServer || activeSort || isReversed;
```

Comment:
The condition `goods !== goodsFromServer` will always be true because `goods` is a new array instance even if it contains the same elements as `goodsFromServer`. Consider using a deep comparison or checking if `activeSort` or `isReversed` is set to determine if the reset button should be visible.